python/lsst/summit/utils/astrometry/fastStarTrackerAnalysis.py

```diff
@@ -0,0 +1,551 @@
+    sctrl.setNumSigmaClip(nSigma)
+    sctrl.setNumIter(2)  # this is always plenty here
+    statTypes = afwMath.MEANCLIP | afwMath.STDEVCLIP
+    stats = afwMath.makeStatistics(exp.maskedImage, statTypes, sctrl)
+    std, _ = stats.getResult(afwMath.STDEVCLIP)
+    mean, _ = stats.getResult(afwMath.MEANCLIP)
+    return mean, std
+
+
+def countOverThresholdPixels(cutout, bgMean, bgStd, nSigma=15):
+    """Get the number of pixels in the cutout which are 'in the source'.
+
+    From the one image I've looked at so far, the drop-off is quite slow
+    probably due to some combination of focus, plate scale, star brightness,
+    pointing quality etc, so the default nSigma is 15 here as that looked about
+    right when I plotted it by eye.
+
+    Parameters
+    ----------
+    cutout : `np.array`
+        The cutout to measure.
+    bgMean : `float`
+        The background level.
+    bgStd : `float`
+        The clipped standard deviation in the image.
+    nSigma : `float`, optional
+        The number of sigma above background at which to count pixels as being
+        over threshold.
+
+    Returns
+    -------
+    nPix : `int`
+        The number of pixels above threshold.
+    """
+    inds = np.where(cutout > (bgMean + 0*bgStd))
+    return len(inds[0])
+
+
+def sortSourcesByFlux(sources, reverse=False):
+    """Sort the sources by flux, returning the brightest first.
+
+    Parameters
+    ----------
+    sources : `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
+        The list of sources to sort.
+    reverse : `bool`, optional
+        Return the brightest at the start of the list if ``reverse`` is
+        ``False``, or the brightest last if ``reverse`` is ``True``.
+
+    Returns
+    -------
+    sources : `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
+        The sources, sorted by flux.
+    """
+    # invert reverse because we want brightest first by default, but want the
+    # reverse arg to still behave as one would expect
+    return sorted(sources, key=lambda s: s.rawFlux, reverse=not reverse)
+
+
+@dataclass(slots=True)
+class Source:
+    """A dataclass for FastStarTracker analysis results.
+    """
+    # raw numbers
+    centroidX: float = np.nan  # in image coordinates
+    centroidY: float = np.nan  # in image coordinates
+    rawFlux: float = np.nan
+    nPix: int = np.nan
+    bbox: geom.Box2I = None
+    cutout: np.array = None
+    localCentroidX: float = np.nan  # in cutout coordinates
+    localCentroidY: float = np.nan  # in cutout coordinates
+
+    # numbers from the hsm moments fit
+    hsmFittedFlux: float = np.nan
+    hsmCentroidX: float = np.nan
+    hsmCentroidY: float = np.nan
+    moments: galsim.hsm.ShapeData = None  # keep the full fit even though we pull some things out too
+
+    imageBackground: float = np.nan
+    imageStddev: float = np.nan
+    nSourcesInImage: int = np.nan
```

Comment:
I definitely paused there, for exactly that reason. I then thought about calling it something like `siblings` because that would make it make sense (and that's how I expect it to be used) but it was too close to deblender terminology, plus that's begging for an off-by-one error too, so I stuck with this. But I certainly agree in principle.